### PR TITLE
Pod Mutating Webhook & Secret Annotation Injector

### PR DESCRIPTION
**Pull request by EngHabu**
# TL;DR
Add a Pod Mutating WebHook cmd for flytepropeller. The current setup only implements a secrets injector that can either inject secrets available to FlytePropeller or k8s secrets.

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [X] Code documentation added
 - [X] Any pending items have an associated Issue

## Complete description
* `flytepropeller webhook init-certs` command issues a CA/Cert/PrivateKey and store them into a secret given a secret name
* `flytepropeller webhook` creates a MutatingWebhookConfigration object in etcd., a webhook server and registers it with API Server. From there:
    1. It starts listening to Pods labeled with `inject-flyte-secrets: true`. 
    2. Parses out annotations on the pod to know which secrets to inject
    3. Lookups the env (envVars and mounted files) for the secret, if found it'll inject it directly into the pod
    4. Adds k8s EnvFrom or VolumeMountSource to mount the right secret into the Pod
* Plugin Manager change to automatically add `inject-flyte-secrets: true` to CRDs if the task has secrets in its TaskTemplate.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/800

### CODE REVIEW

File: cmd/controller/cmd/init_certs.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
+package cmd
```
Comment (kumare3) [inline cmd/controller/cmd/init_certs.go:1]: lets put a top level docs for the intent of this cmd?

File: cmd/controller/cmd/webhook.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
+package cmd
```
Comment (kumare3) [inline cmd/controller/cmd/webhook.go:1]: another package comment?

File: pkg/controller/nodes/task/k8s/plugin_manager.go [hunk 3]
```diff
@@ -176,12 +194,28 @@ func (e *PluginManager) getPodEffectiveResourceLimits(ctx context.Context, pod *
...
+	}
+
+	k8sTaskCtx := newTaskExecutionContext(tCtx, k8sTaskCtxMetadata)
+
+	o, err := e.plugin.BuildResource(ctx, k8sTaskCtx)
 	if err != nil {
 		return pluginsCore.UnknownTransition, err
 	}
 
+<<<<<<< HEAD
```
Comment (kumare3) [inline pkg/controller/nodes/task/k8s/plugin_manager.go:214]: conflict?

File: pkg/controller/nodes/task/k8s/task_exec_context.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
+package k8s
```
Comment (kumare3) [inline pkg/controller/nodes/task/k8s/task_exec_context.go:1]: docs on the top?

File: pkg/webhook/pod.go [hunk 5]
```diff
@@ -0,0 +1,197 @@
+package webhook
```
Comment (kumare3) [inline pkg/webhook/pod.go:1]: definitely need docs here

**Comment by kumare3** [comment]
I do not know whats the best way to do this, but we should have a general flow somewhere for secrets, maybe in the readme? that Propeller will inject secret proto into annotations and then the injector will actually add the secrets to every pod

**Comment by EngHabu** [comment]
too many conflicts... starting a new branch

**Pull request closed without merging**